src/core/shape/frustum.js

```diff
@@ -34,66 +43,22 @@ class Frustum {
      * frustum.setFromMat4(projMat);
      */
     setFromMat4(matrix) {
-
-        const normalize = (plane) => {
-            const t = Math.sqrt(plane[0] * plane[0] + plane[1] * plane[1] + plane[2] * plane[2]);
-            const invT = 1 / t;
-            plane[0] *= invT;
-            plane[1] *= invT;
-            plane[2] *= invT;
-            plane[3] *= invT;
-        };
-
         const vpm = matrix.data;
         const planes = this.planes;
 
-        // Extract the numbers for the RIGHT plane
-        let plane = planes[0];
-        plane[0] = vpm[3] - vpm[0];
-        plane[1] = vpm[7] - vpm[4];
-        plane[2] = vpm[11] - vpm[8];
-        plane[3] = vpm[15] - vpm[12];
-        normalize(plane);
-
-        // Extract the numbers for the LEFT plane
-        plane = planes[1];
-        plane[0] = vpm[3] + vpm[0];
-        plane[1] = vpm[7] + vpm[4];
-        plane[2] = vpm[11] + vpm[8];
-        plane[3] = vpm[15] + vpm[12];
-        normalize(plane);
-
-        // Extract the BOTTOM plane
-        plane = planes[2];
-        plane[0] = vpm[3] + vpm[1];
-        plane[1] = vpm[7] + vpm[5];
-        plane[2] = vpm[11] + vpm[9];
-        plane[3] = vpm[15] + vpm[13];
-        normalize(plane);
-
-        // Extract the TOP plane
-        plane = planes[3];
-        plane[0] = vpm[3] - vpm[1];
-        plane[1] = vpm[7] - vpm[5];
-        plane[2] = vpm[11] - vpm[9];
-        plane[3] = vpm[15] - vpm[13];
-        normalize(plane);
-
-        // Extract the FAR plane
-        plane = planes[4];
-        plane[0] = vpm[3] - vpm[2];
-        plane[1] = vpm[7] - vpm[6];
-        plane[2] = vpm[11] - vpm[10];
-        plane[3] = vpm[15] - vpm[14];
-        normalize(plane);
+        const setPlane = (plane, a, b, c, d) => {
+            plane.normal.set(vpm[3] + a * vpm[0], vpm[7] + b * vpm[4], vpm[11] + c * vpm[8]);
+            plane.distance = vpm[15] + d * vpm[12];
+            plane.normalize();
+        };
 
-        // Extract the NEAR plane
-        plane = planes[5];
-        plane[0] = vpm[3] + vpm[2];
-        plane[1] = vpm[7] + vpm[6];
-        plane[2] = vpm[11] + vpm[10];
-        plane[3] = vpm[15] + vpm[14];
-        normalize(plane);
+        // Extract the numbers for the planes
+        setPlane(planes[0], -1, -1, -1, -1); // RIGHT
```

Comment:
hmm not sure about this last commit?
for example in the original code:
```
plane[0].distance = vpm[15] - vpm[12];
plane[4].distance = vpm[15] - vpm[14];
```

but new code is always
```
plane.distance = vpm[15] + d * vpm[12];
```

but with some optional sign

Reply:
Yep, I messed up. I've reimplemented this. So much nicer! 🚀